### PR TITLE
Core/Vehicle: add a way to delay the Vehicle despawn Time if its needed

### PR DESCRIPTION
**Pull request by gonzo1247**
**Changes proposed:**
-  Some Vehicles need a delayed despawn Time for some action after Player/Creature exit the vehicle. With this PR is this now possible. It's add a new Smart Action 'SMART_ACTION_SET_DELAY_VEHICLE_DESPAWN'  to delayed the despawn in SmartAI.
With the function 'SetTimedVehicleDespawn()' can this also in C++ Code used.
-  
-  

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:**

Closes #  (insert issue tracker number)
https://github.com/TrinityCore/TrinityCore/issues/18089
Now this issue can closed

**Tests performed:**
Build works, if ingame testet with different time with SmartAI and C++
(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by Shauren** [comment]
Wouldnt it be better to introduce a new db table `vehicle_template` that would have a column like that?

**Comment by jackpoz** [comment]
did you miss to push the change where GetTimedVehicleDespawn() is used ? I wonder if the new field should be in Vehicle class itself

**Comment by gonzo1247** [comment]
> 
> 
> did you miss to push the change where GetTimedVehicleDespawn() is used ? I wonder if the new field should be in Vehicle class itself

yea i missed to stage this change. its in the follow commit

**Comment by gonzo1247** [comment]
> 
> 
> Wouldnt it be better to introduce a new db table `vehicle_template` that would have a column like that?

this is also possible - i change it and push it again.
That would have the advantage that the timer is there from the start.

**Comment by gonzo1247** [comment]
Its been changed and works over a new table

File: src/server/game/Entities/Vehicle/Vehicle.cpp [hunk 1]
```diff
@@ -885,3 +885,12 @@ void VehicleJoinEvent::Abort(uint64)
     if (Passenger->IsInWorld() && Passenger->HasUnitTypeMask(UNIT_MASK_ACCESSORY))
         Passenger->ToCreature()->DespawnOrUnsummon();
 }
+
+Milliseconds Vehicle::GetDelayDespawnTime()
+{
+    if (VehicleTemplateList const* temp = sObjectMgr->GetVehicleTemplate(this))
+        for (VehicleTemplateList::const_iterator itr = temp->begin(); itr != temp->end(); ++itr)
```
Comment (Shauren) [inline src/server/game/Entities/Vehicle/Vehicle.cpp:892]: What exactly was your plan by making this a list/vector?

Comment (gonzo1247) [reply src/server/game/Entities/Vehicle/Vehicle.cpp:892]: My first plan was that the hold more as one data - there was with a ForceRespawn Timer but this was discarded and i didn't change the ground part.

**Comment by gonzo1247** [comment]
Or gives a bether/other way to impelemt this?
my idea is, that so can easyer expand if the vehicle_template becomes new fields

**Pull request merged**